### PR TITLE
Add bindingstatus controller

### PR DESCRIPTION
**Pull request by Poor12**
**What type of PR is this?**
/kind cleanup
/kind feature

**What this PR does / why we need it**:
Currently, the synchronization and state collection of binding are both processed in one controller, and there will be redundant calculations, because sometimes only synchronization is required, and sometimes only state collection is required, so I moved the logic of state collection from the binding-controller isolated.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

**Review by jwcesign — COMMENTED** [review]
/lgtm

**Review by RainbowMango — COMMENTED** [review]
Generally looks good to me.
Find some nits from the previous code, I guess this is a good chance to fix them.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

File: pkg/controllers/binding/binding_controller.go [hunk 1]
```diff
@@ -103,19 +97,18 @@ func (c *ResourceBindingController) syncBinding(binding *workv1alpha2.ResourceBi
...
+	workload, err := helper.FetchResourceTemplate(c.DynamicClient, c.InformerManager, c.RESTMapper, binding.Spec.Resource)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			// It might happen when the resource template has been removed but the garbage collector hasn't removed
 			// the ResourceBinding which dependent on resource template.
 			// So, just return without retry(requeue) would save unnecessary loop.
 			return controllerruntime.Result{}, nil
 		}
-		klog.Errorf("Failed to fetch workload for resourceBinding(%s/%s). Error: %v.",
+		klog.Errorf("Failed to fetch workload for resourceBinding(%s/%s). Error: %v",
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply pkg/controllers/binding/binding_controller.go]: cc @XiShanYongYe-Chang

[the remark this answers is not in the document]
Comment (XiShanYongYe-Chang) [reply pkg/controllers/binding/binding_controller.go]: I checked most of the code elsewhere and didn't add the symbol in the similar error output.

Besides, if there is a `.` in err itself, there may be two `.` here.

**Pull request merged**